### ISSUE TITLE
Error: Chanel not Responding

### ISSUE THREAD

**Issue by faithless1108** [Cigaras/IPTV.bundle#74]
Hello Everybody, 

i installed the Plugin, Channel in Plex appeared. I did the configuration and gave the m3u-File. But when I click the channel a Message "Chanel not Responding" appears... 

The Log says:
```
2017-01-31 23:58:12,221 (7f10e5ffb700) :  DEBUG (preferences:258) - Loaded preferences from DefaultPrefs.json
2017-01-31 23:58:12,222 (7f10e5ffb700) :  DEBUG (preferences:178) - Loaded the user preferences for com.plexapp.plugins.iptv
2017-01-31 23:58:12,224 (7f10e5ffb700) :  CRITICAL (runtime:889) - Exception (most recent call last):
  File "/usr/lib/plexmediaserver/Resources/Plug-ins-b38628e/Framework.bundle/Contents/Resources/Versions/2/Python/Framework/components/runtime.py", line 843, in handle_request
    result = f(**d)
  File "/usr/lib/plexmediaserver/Resources/Plug-ins-b38628e/Framework.bundle/Contents/Resources/Versions/2/Python/Framework/handlers/base.py", line 111, in call
    result = self.func(*args, **kwargs)
  File "/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Plug-ins/IPTV.bundle/Contents/Code/__init__.py", line 69, in MainMenu
    return ListItems(unicode(L('All')))
  File "/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Plug-ins/IPTV.bundle/Contents/Code/__init__.py", line 74, in ListItems
    items_list = STREAMS[group].values()
  File "/usr/lib/plexmediaserver/Resources/Plug-ins-b38628e/Framework.bundle/Contents/Resources/Versions/2/Python/Framework/code/sandbox.py", line 108, in <lambda>
    _getitem_           = lambda x, y: x.__getitem__(y),
KeyError: u'All'

2017-01-31 23:58:12,233 (7f10e5ffb700) :  DEBUG (runtime:88) - Sending packed state data (105 bytes)
2017-01-31 23:58:12,233 (7f10e5ffb700) :  DEBUG (runtime:924) - Response: [500] 1179 bytes
```

Do you have any advice?

**Comment by Cigaras** [comment]
I can not reproduce Your issue, can You share Your playlist and plugins settings?

Settings are stored in `..\Plex Media Server\Plug-in Support\Preferences\com.plexapp.plugins.iptv.xml` file.

Also what operating system are You using?

**Comment by faithless1108** [comment]
Hi, 

sure... 

<?xml version='1.0' encoding='utf-8'?>
<PluginPreferences>
  <playlist>tvsd.m3u</playlist>
  <xmltv>guide.xml</xmltv>
  <optimized_for_streaming>false</optimized_for_streaming>
  <guide_hours>8</guide_hours>
  <player_prefs>false</player_prefs>
  <sort_lists>true</sort_lists>
  <items_per_page>100</items_per_page>
  <rtmp>false</rtmp>
  <sort_groups>true</sort_groups>
</PluginPreferences>

**Comment by Cigaras** [comment]
Your playlist has additional lines between #EXTINF and url that I've ever heard of:
```
#EXTM3U
#EXTINF:0,Das Erste
#EXTVLCOPT:network-caching=1000
rtsp://192.168....
#EXTINF:0,BR Fernsehen Süd
#EXTVLCOPT:network-caching=1000
rtsp://192.168....
...
```
[Wikipedia](https://en.wikipedia.org/wiki/M3U) has nothing to say about those #EXTVLCOPT lines, google wasn't of much help either, for the moment best I can suggest is to delete those lines, and I might update the plug-in to skip #EXTVLCOPT automaticaly, however those lines might be required for streams to work however I have no idea how to pass them to Plex player and I really doubt Plex would understand them anyway.

**Comment by faithless1108** [comment]
Hey, 

thanks for the Hint. I removed the Lines u mentioned. Playlist is still working in VLC but sill nocht working in Plex.

Do you have any other Ideas on how to get rid of the Problem?

**Comment by Cigaras** [comment]
I can't play You streems in VLC, I gues they are ment for Your network, and I can not help here any more, Plex is not very good ar playing RTSP streams either way, must be Plex players issue.

**Comment by Sidamos666** [comment]
Hi, 
I have the same problem and the same #EXTVLCOPT lines in my playlist
which was generated btw by a FritzWlan Repeater DVB-C just fyi
maybe it helps someone
I`ll try next with Kodi
thank you for your great plugin anyway!

**Comment by Cigaras** [comment]
Version [2.0.0](https://github.com/Cigaras/IPTV.bundle/tree/beta) (no release yet, You can download [beta.zip](https://github.com/Cigaras/IPTV.bundle/archive/beta.zip) and rename it to IPTV.bundle) is aware of #EXTVLCOPT lines and will not break.

**Comment by faithless1108** [comment]
Hey, 

Thank you for your work. But it is still not working... Here is my logfile...

2017-02-20 22:24:44,378 (7f657eb91700) :  INFO (core:349) - Starting framework core - Version: 2.6.3, Build: b46e0ea (Thu Feb 02 20:49:35 UTC 2017)
2017-02-20 22:24:44,378 (7f657eb91700) :  DEBUG (core:361) - Using the standard policy
2017-02-20 22:24:44,378 (7f657eb91700) :  DEBUG (core:450) - Starting runtime component.
2017-02-20 22:24:44,379 (7f657eb91700) :  DEBUG (core:450) - Starting caching component.
2017-02-20 22:24:44,379 (7f657eb91700) :  DEBUG (core:450) - Starting data component.
2017-02-20 22:24:44,379 (7f657eb91700) :  DEBUG (core:450) - Starting networking component.
2017-02-20 22:24:44,379 (7f657eb91700) :  DEBUG (networking:284) - Loaded HTTP cookies
2017-02-20 22:24:44,380 (7f657eb91700) :  DEBUG (networking:452) - Setting the default network timeout to 20.0
2017-02-20 22:24:44,380 (7f657eb91700) :  DEBUG (core:450) - Starting localization component.
2017-02-20 22:24:44,380 (7f657eb91700) :  INFO (localization:409) - Setting the default locale to en-us
2017-02-20 22:24:44,380 (7f657eb91700) :  DEBUG (localization:427) - Loaded en strings
2017-02-20 22:24:44,381 (7f657eb91700) :  DEBUG (core:450) - Starting messaging component.
2017-02-20 22:24:44,381 (7f657eb91700) :  DEBUG (core:450) - Starting debugging component.
2017-02-20 22:24:44,381 (7f657eb91700) :  DEBUG (core:450) - Starting services component.
2017-02-20 22:24:44,381 (7f657eb91700) :  DEBUG (core:450) - Starting myplex component.
2017-02-20 22:24:44,381 (7f655effd700) :  DEBUG (networking:166) - Requesting 'http://127.0.0.1:32400/system/messaging/clear_events/com.plexapp.plugins.iptv'
2017-02-20 22:24:44,381 (7f657eb91700) :  DEBUG (core:450) - Starting notifications component.
2017-02-20 22:24:44,453 (7f657eb91700) :  DEBUG (accessor:68) - Creating a new model access point for provider com.plexapp.plugins.iptv in namespace 'metadata'
2017-02-20 22:24:44,456 (7f657eb91700) :  DEBUG (networking:166) - Requesting 'http://127.0.0.1:32400/:/plugins/com.plexapp.system/resourceHashes'
2017-02-20 22:24:44,466 (7f655effd700) :  DEBUG (services:265) - Plug-in is not daemonized - loading services from system
2017-02-20 22:24:44,467 (7f657eb91700) :  DEBUG (runtime:1117) - Created a thread named 'load_all_services'
2017-02-20 22:24:44,467 (7f657eb91700) :  DEBUG (runtime:1117) - Created a thread named 'get_server_info'
2017-02-20 22:24:44,467 (7f657eb91700) :  DEBUG (core:150) - Finished starting framework core
2017-02-20 22:24:44,467 (7f657eb91700) :  DEBUG (core:560) - Loading plug-in code
2017-02-20 22:24:44,487 (7f655dffb700) :  DEBUG (networking:166) - Requesting 'http://127.0.0.1:32400'
2017-02-20 22:24:44,491 (7f657eb91700) :  DEBUG (core:566) - Finished loading plug-in code
2017-02-20 22:24:44,491 (7f655effd700) :  DEBUG (networking:166) - Requesting 'http://127.0.0.1:32400/:/plugins/com.plexapp.system/messaging/function/X0J1bmRsZVNlcnZpY2U6QWxsU2VydmljZXM_/Y2VyZWFsMQoxCmxpc3QKMApyMAo_/Y2VyZWFsMQoxCmRpY3QKMApyMAo_'
2017-02-20 22:24:44,533 (7f657eb91700) :  DEBUG (runtime:640) - Adding a prefix handler for 'IPTV' ('/video/iptv')
2017-02-20 22:24:44,536 (7f655dffb700) :  DEBUG (core:538) - Machine identifier is 7e7a2c92efccb0bd84b6991568fabcd321767b44
2017-02-20 22:24:44,536 (7f655dffb700) :  DEBUG (core:539) - Server version is 1.3.4.3285-b46e0ea
2017-02-20 22:24:44,575 (7f655effd700) :  DEBUG (services:362) - Loaded services
2017-02-20 22:24:44,577 (7f655e7fc700) :  DEBUG (services:438) - No shared code to load
2017-02-20 22:24:44,577 (7f657eb91700) :  DEBUG (preferences:258) - Loaded preferences from DefaultPrefs.json
2017-02-20 22:24:44,578 (7f657eb91700) :  DEBUG (preferences:178) - Loaded the user preferences for com.plexapp.plugins.iptv
2017-02-20 22:24:44,578 (7f657eb91700) :  INFO (datakit:118) - No default dictionary file
2017-02-20 22:24:44,579 (7f657eb91700) :  DEBUG (runtime:1117) - Created a thread named 'PlaylistReloader'
2017-02-20 22:24:44,579 (7f657eb91700) :  DEBUG (runtime:1117) - Created a thread named 'GuideReloader'
2017-02-20 22:24:44,579 (7f657eb91700) :  INFO (core:611) - Started plug-in
2017-02-20 22:24:44,579 (7f657eb91700) :  DEBUG (socketinterface:160) - Starting socket server
2017-02-20 22:24:44,580 (7f657eb91700) :  DEBUG (runtime:1117) - Created a thread named 'start'
2017-02-20 22:24:44,580 (7f657eb91700) :  INFO (socketinterface:184) - Socket server started on port 34645
2017-02-20 22:24:44,580 (7f657eb91700) :  INFO (pipeinterface:25) - Entering run loop
2017-02-20 22:24:44,581 (7f657eb91700) :  DEBUG (runtime:717) - Handling request GET /:/prefixes
2017-02-20 22:24:44,581 (7f657eb91700) :  DEBUG (runtime:814) - Found route matching /:/prefixes
2017-02-20 22:24:44,582 (7f657eb91700) :  DEBUG (runtime:924) - Response: [200] MediaContainer, 424 bytes
2017-02-20 22:24:49,581 (7f655e7fc700) :  INFO (datakit:131) - Saved the dictionary file
2017-02-20 22:27:29,039 (7f655e7fc700) :  DEBUG (runtime:717) - Handling request GET /:/plugins/com.plexapp.plugins.iptv/resources/icon-default.png?t=1487625883
2017-02-20 22:27:29,047 (7f655e7fc700) :  DEBUG (runtime:814) - Found route matching /:/plugins/com.plexapp.plugins.iptv/resources/icon-default.png
2017-02-20 22:27:29,049 (7f655e7fc700) :  DEBUG (runtime:88) - Sending packed state data (105 bytes)
2017-02-20 22:27:29,049 (7f655e7fc700) :  DEBUG (runtime:924) - Response: [200] DataObject, 214239 bytes
2017-02-20 22:27:32,174 (7f655e7fc700) :  DEBUG (runtime:717) - Handling request GET /video/iptv/:/prefs
2017-02-20 22:27:32,176 (7f655e7fc700) :  DEBUG (runtime:814) - Found route matching /video/iptv/:/prefs
2017-02-20 22:27:32,183 (7f655e7fc700) :  DEBUG (runtime:88) - Sending packed state data (105 bytes)
2017-02-20 22:27:32,184 (7f655e7fc700) :  DEBUG (runtime:924) - Response: [200] MediaContainer, 1891 bytes
2017-02-20 22:27:37,549 (7f655e7fc700) :  DEBUG (runtime:717) - Handling request GET /video/iptv
2017-02-20 22:27:37,551 (7f655e7fc700) :  DEBUG (runtime:814) - Found route matching /video/iptv
2017-02-20 22:27:37,552 (7f655e7fc700) :  CRITICAL (runtime:889) - Exception (most recent call last):
  File "/usr/lib/plexmediaserver/Resources/Plug-ins-b46e0ea/Framework.bundle/Contents/Resources/Versions/2/Python/Framework/components/runtime.py", line 843, in handle_request
    result = f(**d)
  File "/usr/lib/plexmediaserver/Resources/Plug-ins-b46e0ea/Framework.bundle/Contents/Resources/Versions/2/Python/Framework/handlers/base.py", line 111, in call
    result = self.func(*args, **kwargs)
  File "/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Plug-ins/IPTV.bundle/Contents/Code/__init__.py", line 62, in MainMenu
    return ListItems(unicode(L('All')))
  File "/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Plug-ins/IPTV.bundle/Contents/Code/__init__.py", line 68, in ListItems
    items_list = streams[group].values()
  File "/usr/lib/plexmediaserver/Resources/Plug-ins-b46e0ea/Framework.bundle/Contents/Resources/Versions/2/Python/Framework/code/sandbox.py", line 108, in <lambda>
    _getitem_           = lambda x, y: x.__getitem__(y),
KeyError: u'All'

2017-02-20 22:27:37,552 (7f655e7fc700) :  DEBUG (runtime:88) - Sending packed state data (105 bytes)
2017-02-20 22:27:37,552 (7f655e7fc700) :  DEBUG (runtime:924) - Response: [500] 1179 bytes
2017-02-20 22:27:48,314 (7f655e7fc700) :  DEBUG (runtime:717) - Handling request GET /video/iptv/:/prefs
2017-02-20 22:27:48,316 (7f655e7fc700) :  DEBUG (runtime:49) - Received packed state data (80 bytes)
2017-02-20 22:27:48,318 (7f655e7fc700) :  DEBUG (runtime:814) - Found route matching /video/iptv/:/prefs
2017-02-20 22:27:48,325 (7f655e7fc700) :  DEBUG (runtime:88) - Sending packed state data (105 bytes)
2017-02-20 22:27:48,325 (7f655e7fc700) :  DEBUG (runtime:924) - Response: [200] MediaContainer, 1891 bytes

**Comment by Cigaras** [comment]
I need to see Your playlist.

**Comment by faithless1108** [comment]
Hey, 

Here it is:
[tvsd.m3u.txt](https://github.com/Cigaras/IPTV.bundle/files/797111/tvsd.m3u.txt)

**Comment by Sidamos666** [comment]
[playlist_SANI.m3u.txt](https://github.com/Cigaras/IPTV.bundle/files/797122/playlist_SANI.m3u.txt)
and here is mine
but I deleted the #EXTVLCOPT lines

btw that playlist works flawlessly in Kodi with IPTV Simple Client 
just fyi

**Comment by Cigaras** [comment]
Fixed in [version 2.0.3](https://github.com/Cigaras/IPTV.bundle/releases)
